validator/client/metrics.go

```diff
@@ -161,66 +161,109 @@ func (v *validator) LogValidatorGainsAndLosses(ctx context.Context, slot uint64)
+				"percentChangeSinceStart": fmt.Sprintf("%.5f%%", percentSinceStart*100),
 			}).Info("Previous epoch voting summary")
 			if v.emitAccountMetrics {
 				ValidatorBalancesGaugeVec.WithLabelValues(fmtKey).Set(newBalance)
 			}
 		}
+		v.prevBalance[pubKeyBytes] = resp.BalancesBeforeEpochTransition[i]
+	}
+
+	v.UpdateLogAggregateStats(resp, slot)
+	return nil
+}
 
+//UpdateLogAggregateStats updates and logs the voteStats struct of a validator using the RPC response obtained during LogValidatorGainsAndLosses
+func (v *validator) UpdateLogAggregateStats(resp *ethpb.ValidatorPerformanceResponse, slot uint64) {
+	summary := &v.voteStats
+	currentEpoch := slot / params.BeaconConfig().SlotsPerEpoch
+	var included, correctSource, correctTarget, correctHead int
+
+	for i := range resp.PublicKeys {
 		if resp.InclusionSlots[i] != ^uint64(0) {
 			included++
+			summary.includedAttestedCount++
+			summary.totalDistance += resp.InclusionDistances[i]
 		}
 		if resp.CorrectlyVotedSource[i] {
-			votedSource++
+			correctSource++
+			summary.correctSources++
 		}
 		if resp.CorrectlyVotedTarget[i] {
-			votedTarget++
+			correctTarget++
+			summary.correctTargets++
 		}
 		if resp.CorrectlyVotedHead[i] {
-			votedHead++
+			correctHead++
+			summary.correctHeads++
 		}
-		v.prevBalance[pubKeyBytes] = resp.BalancesBeforeEpochTransition[i]
 	}
 
+	summary.totalAttestedCount += uint64(len(resp.InclusionSlots))
+	summary.totalSources += uint64(included)
+	summary.totalTargets += uint64(included)
+	summary.totalHeads += uint64(included)
+
 	log.WithFields(logrus.Fields{
-		"epoch":                          prevEpoch,
-		"attestationInclusionPercentage": fmt.Sprintf("%.0f%%", (float64(included)/float64(len(resp.InclusionSlots)))*100),
-		"correctlyVotedSourcePercentage": fmt.Sprintf("%.0f%%", (float64(votedSource)/float64(len(resp.CorrectlyVotedSource)))*100),
-		"correctlyVotedTargetPercentage": fmt.Sprintf("%.0f%%", (float64(votedTarget)/float64(len(resp.CorrectlyVotedTarget)))*100),
-		"correctlyVotedHeadPercentage":   fmt.Sprintf("%.0f%%", (float64(votedHead)/float64(len(resp.CorrectlyVotedHead)))*100),
+		"epoch":                   currentEpoch - 1,
+		"attestationInclusionPct": fmt.Sprintf("%.0f%%", (float64(included)/float64(len(resp.InclusionSlots)))*100),
+		"correctlyVotedSourcePct": fmt.Sprintf("%.0f%%", (float64(correctSource)/float64(included))*100),
+		"correctlyVotedTargetPct": fmt.Sprintf("%.0f%%", (float64(correctTarget)/float64(included))*100),
+		"correctlyVotedHeadPct":   fmt.Sprintf("%.0f%%", (float64(correctHead)/float64(included))*100),
 	}).Info("Previous epoch aggregated voting summary")
 
-	return nil
+	var totalStartBal, totalPrevBal uint64
+	for i, val := range v.startBalances {
+		totalStartBal += val
+		totalPrevBal += v.prevBalance[i]
+	}
+
+	log.WithFields(logrus.Fields{
+		"numberOfEpochs":           fmt.Sprintf("%d", currentEpoch-summary.startEpoch+1), //+1 to include the original epoch
```

Comment:
I don't think we need to +1 here. The process epoch -> voting summary is always an epoch behind

Reply:
Addressed in 1347c82.  Thanks